apicast/src/configuration.lua

```diff
@@ -478,20 +478,20 @@ function _M.curl(endpoint)
 
   url = concat({ scheme, '://', concat({user or '', pass or ''}, ':'), '@', host, path or '/admin/api/nginx/spec.json' }, '')
 
-  local config, exit, code = util.system('curl --silent --show-error --fail --max-time ' .. timeout .. ' --location ' .. url)
+  local config, stderr, code = util.system('curl --silent --show-error --fail --max-time ' .. timeout .. ' --location ' .. url)
 
-  ngx.log(ngx.INFO, 'configuration request sent: ' .. url)
+  ngx.log(ngx.INFO, 'configuration request sent: ', url)
 
-  if config then
-    ngx.log(ngx.DEBUG, 'configuration response received:' .. config)
+  if len(config) > 0 then
+    ngx.log(ngx.DEBUG, 'configuration response received:', config)
     return config
   else
     if code then
-      ngx.log(ngx.ERR, 'configuration download error ' .. exit .. ' ' .. code)
-      return nil, 'curl fished with ' .. exit .. ' ' .. code
+      ngx.log(ngx.ERR, 'configuration download error ', stderr, ' ', code)
+      return nil, 'curl fished with ' .. stderr .. ' ' .. code
     else
-      ngx.log(ngx.WARN, 'configuration download error: ' .. exit)
-      return nil, exit
+      ngx.log(ngx.WARN, 'configuration download error: ',  stderr)
```

Comment:
Because we don't really have an information why. The code is missing so this might never happen. 

But to be honest. I really don't know :)